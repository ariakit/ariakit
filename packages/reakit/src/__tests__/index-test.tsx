import "jest-styled-components";
import * as React from "react";
import { render } from "react-testing-library";
import * as c from "..";

jest.spyOn(console, "error").mockImplementation((...args) => {
  if (/Failed prop type/.test(args[0])) return;
  // eslint-disable-next-line no-console
  console.error(...args);
});

const componentsToTest = [
  c.Avatar,
  c.Backdrop,
  c.Block,
  c.Blockquote,
  c.Box,
  c.Button,
  c.Card,
  c.Card.Fit,
  c.Code,
  c.Divider,
  c.Field,
  c.Flex,
  c.Grid,
  c.Group,
  // @ts-ignore
  c.Group.Item,
  c.Heading,
  c.Hidden,
  c.Hidden.Hide,
  c.Hidden.Show,
  c.Hidden.Toggle,
  c.Image,
  c.Inline,
  c.InlineBlock,
  c.InlineFlex,
  c.Input,
  c.Label,
  c.Link,
  c.List,
  c.Navigation,
  c.Overlay,
  c.Overlay.Hide,
  c.Overlay.Show,
  c.Overlay.Toggle,
  c.Paragraph,
  c.Popover,
  // @ts-ignore
  c.Popover.Arrow,
  // @ts-ignore
  c.Popover.Hide,
  // @ts-ignore
  c.Popover.Show,
  // @ts-ignore
  c.Popover.Toggle,
  c.Portal,
  c.Sidebar,
  c.Sidebar.Hide,
  c.Sidebar.Show,
  c.Sidebar.Toggle,
  c.Step,
  // @ts-ignore
  c.Step.Hide,
  // @ts-ignore
  c.Step.Next,
  // @ts-ignore
  c.Step.Previous,
  // @ts-ignore
  c.Step.Show,
  // @ts-ignore
  c.Step.Toggle,
  c.Table,
  c.Table.Wrapper,
  c.Tabs,
  // @ts-ignore
  c.Tabs.Next,
  // @ts-ignore
  c.Tabs.Panel,
  // @ts-ignore
  c.Tabs.Previous,
  // @ts-ignore
  c.Tabs.Tab,
  c.Toolbar,
  // @ts-ignore
  c.Toolbar.Content,
  // @ts-ignore
  c.Toolbar.Focusable,
  c.Tooltip,
  // @ts-ignore
  c.Tooltip.Arrow
];

componentsToTest.forEach(Component => {
  test(Component.displayName, () => {
    const theme = {
      palette: {
        primary: ["darkred", "red"],
        primaryText: "white"
      }
    };
    const { container } = render(
      <Component theme={theme} absolute palette="primary" tone={1} opaque />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
