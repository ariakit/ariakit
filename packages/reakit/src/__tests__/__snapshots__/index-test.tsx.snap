// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar.as(img) 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<img
  class="c0"
/>
`;

exports[`Backdrop 1`] = `
.c2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c2.c2.c2 {
  position: absolute;
}

.c1 {
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
}

.c1[aria-hidden="true"] {
  pointer-events: none;
  display: none !important;
}

.c0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 998;
  -webkit-tap-highlight-color: rgba(0,0,0,0);
  -moz-tap-highlight-color: rgba(0,0,0,0);
}

<div
  aria-hidden="true"
  class="c0 c1 c2"
  hidden=""
  role="button"
  tabindex="-1"
/>
`;

exports[`Block 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c1.c1.c1 {
  position: absolute;
}

.c0 {
  display: block;
}

<div
  class="c0 c1"
/>
`;

exports[`Blockquote.as(blockquote) 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<blockquote
  class="c0"
/>
`;

exports[`Box.as(div) 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<div
  class="c0"
/>
`;

exports[`Button.as(button) 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<button
  class="c0"
/>
`;

exports[`Card 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c1.c1.c1 {
  position: absolute;
}

.c0 {
  display: inline-block;
}

.c0.c0 > *:not(.c2) {
  margin: 1rem;
}

<div
  class="c0 c1"
/>
`;

exports[`CardFit 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c1.c1.c1 {
  position: absolute;
}

.c2.c2 > *:not(.c0) {
  margin: 1rem;
}

<div
  class="c0 c1"
/>
`;

exports[`Code.as(Component) 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<code
  class="c0"
/>
`;

exports[`Divider.as(hr) 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c1.c1.c1 {
  position: absolute;
}

.c0 {
  border-color: currentColor;
  border-style: solid;
  opacity: 0.2;
  margin: 1em 0;
  height: 0;
  border-width: 1px 0 0 0;
}

<hr
  class="c0 c1"
/>
`;

exports[`Field 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<div
  class="c0"
/>
`;

exports[`Flex 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c1.c1.c1 {
  position: absolute;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  class="c0 c1"
/>
`;

exports[`Grid 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c1.c1.c1 {
  position: absolute;
}

.c0 {
  display: grid;
}

<div
  class="c0 c1"
/>
`;

exports[`Group 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c1.c1.c1 {
  position: absolute;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c0 > *:not(:first-child):not(:last-child),
.c0 > *:not(:first-child):not(:last-child) .GroupItem-kbprqg-0 {
  border-radius: 0;
  border-left-width: 0;
}

.c0 > *:first-child,
.c0 > *:first-child .GroupItem-kbprqg-0 {
  border-bottom-right-radius: 0;
  border-top-right-radius: 0;
}

.c0 > *:last-child,
.c0 > *:last-child .GroupItem-kbprqg-0 {
  border-top-left-radius: 0;
  border-left-width: 0;
  border-bottom-left-radius: 0;
}

<div
  class="c0 c1"
  role="group"
/>
`;

exports[`GroupItem 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c1.c1.c1 {
  position: absolute;
}

.c2 > *:not(:first-child):not(:last-child),
.c2 > *:not(:first-child):not(:last-child) .c0 {
  border-radius: 0;
  border-left-width: 0;
}

.c2 > *:first-child,
.c2 > *:first-child .c0 {
  border-bottom-right-radius: 0;
  border-top-right-radius: 0;
}

.c2 > *:last-child,
.c2 > *:last-child .c0 {
  border-top-left-radius: 0;
  border-left-width: 0;
  border-bottom-left-radius: 0;
}

<div
  class="c0 c1"
/>
`;

exports[`Heading.as(h1) 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<h1
  class="c0"
/>
`;

exports[`Hidden 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c1.c1.c1 {
  position: absolute;
}

.c0 {
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
}

.c0[aria-hidden="true"] {
  pointer-events: none;
  display: none !important;
}

<div
  aria-hidden="true"
  class="c0 c1"
  hidden=""
/>
`;

exports[`HiddenHide.as(button) 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<button
  class="c0"
/>
`;

exports[`HiddenShow.as(button) 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<button
  class="c0"
/>
`;

exports[`HiddenToggle.as(button) 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<button
  class="c0"
/>
`;

exports[`Image.as(img) 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<img
  class="c0"
/>
`;

exports[`Inline.as(span) 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c1.c1.c1 {
  position: absolute;
}

.c0 {
  display: inline;
}

<span
  class="c0 c1"
/>
`;

exports[`InlineBlock 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c1.c1.c1 {
  position: absolute;
}

.c0 {
  display: inline-block;
}

<div
  class="c0 c1"
/>
`;

exports[`InlineFlex 1`] = `
.c2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c2.c2.c2 {
  position: absolute;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<div
  class="c0 c1 c2"
/>
`;

exports[`Input.as(input) 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<input
  class="c0"
  type="text"
/>
`;

exports[`Label.as(label) 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c1.c1.c1 {
  position: absolute;
}

.c0 {
  display: inline-block;
}

<label
  class="c0 c1"
/>
`;

exports[`Link.as(a) 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<a
  class="c0"
/>
`;

exports[`List.as(ul) 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<ul
  class="c0"
/>
`;

exports[`Navigation.as(nav) 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<nav
  class="c0"
/>
`;

exports[`Overlay 1`] = `
.c2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c2.c2.c2 {
  position: absolute;
}

.c1 {
  -webkit-transform: translate3d(-50%,-50%,0px);
  -ms-transform: translate3d(-50%,-50%,0px);
  transform: translate3d(-50%,-50%,0px);
}

.c1[aria-hidden="true"] {
  pointer-events: none;
  display: none !important;
}

.c0 {
  position: fixed;
  z-index: 19900410;
  left: 50%;
  top: 50%;
}

<div
  aria-hidden="true"
  aria-modal="true"
  class="c0 c1 c2"
  hidden=""
  role="dialog"
/>
`;

exports[`OverlayHide 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<button
  class="c0"
/>
`;

exports[`OverlayShow 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<button
  class="c0"
/>
`;

exports[`OverlayToggle 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<button
  class="c0"
/>
`;

exports[`Paragraph.as(p) 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<p
  class="c0"
/>
`;

exports[`Popover.as(div) 1`] = `
.c2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c2.c2.c2 {
  position: absolute;
}

.c1 {
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
}

.c1[aria-hidden="true"] {
  pointer-events: none;
  display: none !important;
}

.c0 {
  position: absolute;
  top: 0;
  left: 0;
  -webkit-user-select: auto;
  -moz-user-select: auto;
  -ms-user-select: auto;
  user-select: auto;
  cursor: auto;
  z-index: 999;
  will-change: transform;
}

<div
  aria-hidden="true"
  class="c0 c1 c2"
  data-placement="bottom"
  hidden=""
  role="group"
/>
`;

exports[`PopoverArrow 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c1.c1.c1 {
  position: absolute;
}

.c0 {
  position: absolute;
  font-size: 30px;
  width: 1em;
  height: 1em;
  pointer-events: none;
  background-color: transparent;
}

.c0 .stroke {
  fill: transparent;
}

.c0 .fill {
  fill: red;
}

[data-placement^="top"] > .c0 {
  top: 100%;
  -webkit-transform: rotateZ(180deg);
  -ms-transform: rotateZ(180deg);
  transform: rotateZ(180deg);
}

[data-placement^="right"] > .c0 {
  right: 100%;
  -webkit-transform: rotateZ(-90deg);
  -ms-transform: rotateZ(-90deg);
  transform: rotateZ(-90deg);
}

[data-placement^="bottom"] > .c0 {
  bottom: 100%;
  -webkit-transform: rotateZ(360deg);
  -ms-transform: rotateZ(360deg);
  transform: rotateZ(360deg);
}

[data-placement^="left"] > .c0 {
  left: 100%;
  -webkit-transform: rotateZ(90deg);
  -ms-transform: rotateZ(90deg);
  transform: rotateZ(90deg);
}

<div
  class="c0 c1"
>
  <svg
    viewBox="0 0 30 30"
  >
    <path
      class="stroke"
      d="M23.7,27.1L17,19.9C16.5,19.3,15.8,19,15,19s-1.6,0.3-2.1,0.9l-6.6,7.2C5.3,28.1,3.4,29,2,29h26 C26.7,29,24.6,28.1,23.7,27.1z"
    />
    <path
      class="fill"
      d="M23,27.8c1.1,1.2,3.4,2.2,5,2.2h2H0h2c1.7,0,3.9-1,5-2.2l6.6-7.2c0.7-0.8,2-0.8,2.7,0L23,27.8L23,27.8z"
    />
  </svg>
</div>
`;

exports[`PopoverHide.as(button) 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<button
  aria-haspopup="true"
  class="c0"
/>
`;

exports[`PopoverShow.as(button) 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<button
  aria-haspopup="true"
  class="c0"
/>
`;

exports[`PopoverToggle.as(button) 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<button
  aria-haspopup="true"
  class="c0"
/>
`;

exports[`Portal 1`] = `null`;

exports[`Sidebar 1`] = `
.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c3.c3.c3 {
  position: absolute;
}

.c2 {
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
}

.c2[aria-hidden="true"] {
  pointer-events: none;
  display: none !important;
}

.c1 {
  position: fixed;
  z-index: 19900410;
  left: 50%;
  top: 50%;
}

.c0 {
  top: 0;
  height: 100vh;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
  overflow: auto;
  left: 0;
  right: auto;
}

<div
  aria-hidden="true"
  aria-modal="true"
  class="c0 c1 c2 c3"
  hidden=""
  role="dialog"
/>
`;

exports[`SidebarHide 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<button
  class="c0"
/>
`;

exports[`SidebarShow 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<button
  class="c0"
/>
`;

exports[`SidebarToggle 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<button
  class="c0"
/>
`;

exports[`Step 1`] = `null`;

exports[`StepHide 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<button
  class="c0"
/>
`;

exports[`StepNext.as(button) 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<button
  class="c0"
/>
`;

exports[`StepPrevious.as(button) 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<button
  class="c0"
/>
`;

exports[`StepShow.as(button) 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<button
  class="c0"
/>
`;

exports[`StepToggle.as(button) 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<button
  class="c0"
/>
`;

exports[`Table.as(table) 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<table
  class="c0"
/>
`;

exports[`TableWrapper 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c1.c1.c1 {
  position: absolute;
}

.c0 {
  max-width: 100%;
  overflow-x: auto;
  -webkit-overflow-scrolling: touch;
}

<div
  class="c0 c1"
/>
`;

exports[`Tabs.as(ul) 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<ul
  class="c0"
  role="tablist"
/>
`;

exports[`TabsNext 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<button
  class="c0"
/>
`;

exports[`TabsPanel 1`] = `null`;

exports[`TabsPrevious 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<button
  class="c0"
/>
`;

exports[`TabsTab.as(li) 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c1.c1.c1 {
  position: absolute;
}

.c0 {
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
}

.c0[aria-hidden="true"] {
  pointer-events: none;
  display: none !important;
}

<li
  aria-controls="undefinedPanel"
  aria-hidden="false"
  class="c0 c1"
  id="undefinedTab"
  role="tab"
  tabindex="-1"
/>
`;

exports[`Toolbar 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c1.c1.c1 {
  position: absolute;
}

.c0 {
  position: relative;
  display: grid;
  width: 100%;
  padding: 8px;
  grid-gap: 8px;
  grid-template: "start center end" / 1fr auto 1fr;
}

.c0[aria-orientation="vertical"] {
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  height: 100%;
  grid-template: "start" 1fr "center" auto "end" 1fr;
}

<div
  aria-orientation="horizontal"
  class="c0 c1"
  role="toolbar"
/>
`;

exports[`ToolbarContent 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c1.c1.c1 {
  position: absolute;
}

.c0 {
  display: grid;
  grid-auto-flow: column;
  grid-auto-columns: min-content;
  grid-gap: inherit;
  grid-area: start;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

[aria-orientation="vertical"] > .c0 {
  grid-auto-flow: row;
  grid-auto-rows: min-content;
  -webkit-box-pack: initial;
  -webkit-justify-content: initial;
  -ms-flex-pack: initial;
  justify-content: initial;
  -webkit-align-content: start;
  -ms-flex-line-pack: start;
  align-content: start;
}

<div
  class="c0 c1"
/>
`;

exports[`ToolbarFocusable 1`] = `
.c0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c0.c0.c0 {
  position: absolute;
}

<div
  class="c0"
  tabindex="-1"
/>
`;

exports[`Tooltip 1`] = `
.c2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c2.c2.c2 {
  position: absolute;
}

.c1 {
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
}

.c1[aria-hidden="true"] {
  pointer-events: none;
  display: none !important;
}

.c0 {
  position: absolute;
  top: 0;
  left: 0;
  -webkit-user-select: auto;
  -moz-user-select: auto;
  -ms-user-select: auto;
  user-select: auto;
  cursor: auto;
  z-index: 999;
  will-change: transform;
}

<div
  aria-hidden="true"
  class="c0 c1 c2"
  data-placement="top"
  hidden=""
  role="tooltip"
/>
`;

exports[`TooltipArrow 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: red;
  color: white;
}

.c1.c1.c1 {
  position: absolute;
}

.c0 {
  position: absolute;
  font-size: 30px;
  width: 1em;
  height: 1em;
  pointer-events: none;
  background-color: transparent;
}

.c0 .stroke {
  fill: transparent;
}

.c0 .fill {
  fill: red;
}

[data-placement^="top"] > .c0 {
  top: 100%;
  -webkit-transform: rotateZ(180deg);
  -ms-transform: rotateZ(180deg);
  transform: rotateZ(180deg);
}

[data-placement^="right"] > .c0 {
  right: 100%;
  -webkit-transform: rotateZ(-90deg);
  -ms-transform: rotateZ(-90deg);
  transform: rotateZ(-90deg);
}

[data-placement^="bottom"] > .c0 {
  bottom: 100%;
  -webkit-transform: rotateZ(360deg);
  -ms-transform: rotateZ(360deg);
  transform: rotateZ(360deg);
}

[data-placement^="left"] > .c0 {
  left: 100%;
  -webkit-transform: rotateZ(90deg);
  -ms-transform: rotateZ(90deg);
  transform: rotateZ(90deg);
}

<div
  class="c0 c1"
>
  <svg
    viewBox="0 0 30 30"
  >
    <path
      class="stroke"
      d="M23.7,27.1L17,19.9C16.5,19.3,15.8,19,15,19s-1.6,0.3-2.1,0.9l-6.6,7.2C5.3,28.1,3.4,29,2,29h26 C26.7,29,24.6,28.1,23.7,27.1z"
    />
    <path
      class="fill"
      d="M23,27.8c1.1,1.2,3.4,2.2,5,2.2h2H0h2c1.7,0,3.9-1,5-2.2l6.6-7.2c0.7-0.8,2-0.8,2.7,0L23,27.8L23,27.8z"
    />
  </svg>
</div>
`;
